### PR TITLE
Use RouteReplace instead of RouteAdd

### PR DESCRIPTION
**Pull request by mogren**
Related to #367 (Comment by [@2intl](https://github.com/aws/amazon-vpc-cni-k8s/commit/33e0e91e2b1afaae2305a3a0b14507c06d1fa562#r32989230))

*Description of changes:*
RouteReplace will create it if it is not already there. Also cleaned up
some debug logs and fixed mocking and tests.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: plugins/routed-eni/driver/driver.go [hunk 1]
```diff
@@ -251,10 +247,13 @@ func setupNS(hostVethName string, contVethName string, netnsPath string, addr *n
 				podRule.Priority = fromContainerRulePriority
 
 				err = netLink.RuleAdd(podRule)
-				// TODO: Replace?
-				if err != nil {
-					log.Errorf("Failed to add pod IP rule: %v", err)
-					return errors.Wrapf(err, "setupNS: failed to add pod rule")
+				if isRuleExistsError(err) {
+					log.Warn("Rule already exists!")
```
Comment (tiffanyfay) [inline plugins/routed-eni/driver/driver.go]: Should we show what rule?

File: plugins/routed-eni/driver/driver.go [hunk 2]
```diff
@@ -251,10 +247,13 @@ func setupNS(hostVethName string, contVethName string, netnsPath string, addr *n
...
+				} else {
+					if err != nil {
+						log.Errorf("Failed to add pod IP rule: %v", err)
```
Comment (tiffanyfay) [inline plugins/routed-eni/driver/driver.go]: What rule? This and next line.

File: plugins/routed-eni/driver/driver_test.go [hunk 3]
```diff
@@ -417,7 +417,7 @@ func TestRunErrNeighAdd(t *testing.T) {
 		mockContVeth.EXPECT().Attrs().Return(mockLinkAttrs),
 		// hostVethMAC
 		mockHostVeth.EXPECT().Attrs().Return(mockLinkAttrs),
-		mockNetLink.EXPECT().NeighAdd(gomock.Any()).Return(errors.New("Error NeighAdd")),
+		mockNetLink.EXPECT().NeighAdd(gomock.Any()).Return(errors.New("error NeighAdd")),
```
Comment (tiffanyfay) [inline plugins/routed-eni/driver/driver_test.go]: error on -- for consistency?

File: plugins/routed-eni/driver/driver_test.go [hunk 4]
```diff
@@ -503,7 +503,7 @@ func TestSetupPodNetwork(t *testing.T) {
 	mockHostVeth.EXPECT().Attrs().Return(mockLinkAttrs)
 	//add host route
 	mockHostVeth.EXPECT().Attrs().Return(mockLinkAttrs)
-	mockNetLink.EXPECT().RouteAdd(gomock.Any()).Return(nil)
+	mockNetLink.EXPECT().RouteReplace(gomock.Any()).Return(nil)
```
Comment (tiffanyfay) [inline plugins/routed-eni/driver/driver_test.go]: Should we add another RouteReplace to show that it works if it exists?

File: plugins/routed-eni/driver/driver_test.go [hunk 5]
```diff
@@ -687,7 +683,7 @@ func TestTearDownPodNetwork(t *testing.T) {
 		IP:   net.ParseIP(testIP),
 		Mask: net.IPv4Mask(255, 255, 255, 255),
 	}
-	tearDownNS(addr, 0, mockNetLink)
+	_ = tearDownNS(addr, 0, mockNetLink)
```
Comment (tiffanyfay) [inline plugins/routed-eni/driver/driver_test.go]: We probably shouldn't ignore an error. Same with line 714

File: plugins/routed-eni/driver/driver_test.go (hunk 4 above)
Comment (mogren) [reply plugins/routed-eni/driver/driver_test.go]: We should improve the tests over all, but right now I'm ok with just making sure that it gets called.

File: plugins/routed-eni/driver/driver_test.go (hunk 5 above)
Comment (mogren) [reply plugins/routed-eni/driver/driver_test.go]: Good point, fixed the test.

**Pull request merged**